common.c

```diff
@@ -0,0 +1,46 @@
+#include <stdlib.h>
+#include <string.h>
+
+#include "common.h"
+
+char *get_line_from_file(FILE *fp)
+{
+	size_t len = 0;
+	char * line = NULL;
+
+	if (getline(&line, &len, fp) == -1) {
+		free(line);
+		return NULL;
+	}
+
+	len = strlen(line);
+	if (len > 0 && line[len - 1] == '\n')
+		line[len - 1] = '\0'; /* trim trailing newline */
+	return line;
+}
+
+char *join_strings(char *const strings[], char *const sep)
```

Comment:
mmm, your `const` are misplaced. Here the pointers are `const`, but we don't really care, we want the content to be `const` since they could be statically-allocated strings.

Reply:
eek. how do I handle the spacing for a `char const * const * const foo;`?

Reply:
eek. how do I handle the spacing around the asterisks for `char const * const * const foo;`?

Reply:
What's wrong with `const char * const strings[]`?

Reply:
Looking at `execve(2)`, they put it as `char *const argv[]`, so let's put the `*` like this, but with the extra `const`.